frontend/src/components/pages/cabins/Admin/DeleteBookingDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import { ApolloQueryResult, OperationVariables, useMutation } from "@apollo/client";
+import { DELETE_BOOKING, SEND_EMAIL } from "@graphql/cabins/mutations";
+import { QUERY_ADMIN_ALL_BOOKINGS } from "@graphql/cabins/queries";
+import { BookingFromQuery } from "@interfaces/cabins";
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  TextField,
+  DialogActions,
+  Button,
+} from "@material-ui/core";
+import { getDecisionEmailProps } from "@utils/cabins";
+import { useState } from "react";
+
+type DialogProps = {
+  bookingToBeDeleted?: BookingFromQuery;
+  setBookingToBeDeleted: any;
```

Comment:
Can we type this more accurately?

Reply:
Wait, can we not type this more accurately?
```typescript
setBookingToBeDeleted: React.Dispatch<React.SetStateAction<BookingFromQuery | undefined>>
```

Reply:
@hermanholmoy

Reply:
Yeah I agree, but it's currently `any`

Reply:
Correct me if I'm wrong but I think this was resolved by aa2f456e5cfd4c85f3683c306a29b00f05b06770 :)